### PR TITLE
Add CircleCI integration

### PR DESCRIPTION
**Pull request by ohlsont**
The CI will check markdown lint and CircleCI config file linting
before merging any code.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+$(PRETTIER):
+	npm install --no-save --no-audit --prefix $(PRETTIER_DIR) prettier@$(PRETTIER_VERSION)
+	chmod +x $@
+	touch $@
+
+MARKDOWNLINT_VERSION := 0.16.0
+MARKDOWNLINT_DIR := $(FILES_DIR)/markdownlint/$(MARKDOWNLINT_VERSION)
+MARKDOWNLINT := $(MARKDOWNLINT_DIR)/node_modules/.bin/markdownlint
+
+$(MARKDOWNLINT):
```
Comment (odsod) [inline Makefile]: Remove this since prettier is used for linting

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	npm install --no-save --no-audit --prefix $(MARKDOWNLINT_DIR) markdownlint-cli@$(MARKDOWNLINT_VERSION)
+	touch $@
+
+# markdown-lint: lint Markdown files with markdownlint
+.PHONY: markdown-lint
+markdown-lint: $(PRETTIER)
+	$(PRETTIER) --check *.md  --parser markdown
```
Comment (odsod) [inline Makefile]: do you need `**/*.md` to check recursively?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# all: all tasks required for a complete build
+.PHONY: all
+all: \
+	circleci-config-validate \
+	markdown-lint
+
+BUILD_DIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
```
Comment (odsod) [inline Makefile]: use `build` as the build folder - this looks like it will generate a `files` dir in the root dir

File: Makefile (hunk 1 above)
Comment (ohlsont) [reply Makefile]: true

**Pull request merged**